playwright/pages/SettingsPage.ts

```diff
@@ -0,0 +1,76 @@
+import { type Locator, type Page, expect } from "@playwright/test";
+
+import { GitRepositoryConnectDialog } from "./components/GitRepositoryConnectDialog";
+import { GitRepositoryDisconnectDialog } from "./components/GitRepositoryDisconnectDialog";
+import { SliceMachinePage } from "./SliceMachinePage";
+
+export class SettingsPage extends SliceMachinePage {
+  readonly breadcrumbLabel: Locator;
+  readonly installGitHubButton: Locator;
+  readonly gitRepositoryConnectDialog: GitRepositoryConnectDialog;
+  readonly gitRepositoryDisconnectDialog: GitRepositoryDisconnectDialog;
+  readonly unauthenticatedErrorTitle: Locator;
+  readonly unauthorizedErrorTitle: Locator;
+  readonly unknownErrorTitle: Locator;
+
+  constructor(page: Page) {
+    super(page);
+
+    /**
+     * Components
+     */
+
+    /**
+     * Static locators
+     */
+    this.breadcrumbLabel = this.breadcrumb.getByText("Settings", {
+      exact: true,
+    });
+    this.installGitHubButton = page.getByRole("button", {
+      name: "GitHub",
+      exact: true,
+    });
+    this.gitRepositoryConnectDialog = new GitRepositoryConnectDialog(page);
+    this.gitRepositoryDisconnectDialog = new GitRepositoryDisconnectDialog(
+      page,
+    );
+    this.unauthenticatedErrorTitle = page.getByText(
+      "It seems like you are logged out",
+      { exact: true },
+    );
+    this.unauthorizedErrorTitle = page.getByText(
+      "It seems like you do not have permission",
+      { exact: true },
+    );
+    this.unknownErrorTitle = page.getByText("Unable to fetch data", {
+      exact: true,
+    });
+  }
+
+  /**
+   * Dynamic locators
+   */
+  getConnectGitRepoButton(name: string): Locator {
+    return this.page
+      .locator("div", { hasText: name })
```

Comment:
These two locators feels a bit off. Sorry I don't have time to look for the solution, but I think it should be closer to what we are doing in `TypesTablePage` for the `getRow` function.

No usage of implementation details should be done in tests. Here "div" for example. Normally we should almost never need `locator`. I added a best practice in the test README for that ^^